rfcs/0001-authentication/text.md

```diff
@@ -0,0 +1,383 @@
+1. [CXG6] The cellxgene server will enforce the authentication for routes that
+   require authentication by checking for the ID token, and rejecting requests
+   that lack a token or have an invalid/expired token. The signature needs to be
+   validated to ensure that the token has not been tampered with. An expired ID
+   token can be refreshed (as long as the refresh token has not expired).
+1. [CXG7] The cellxgene server will have an endpoint to check if the requestor
+   is authenticated. The current (“config”) endpoint can serve this purpose by
+   having an authentication section.
+1. [CXG8] To avoid causing the ID token to expire while the user is working
+   (and thus disrupting the user’s work), the token can be automatically refreshed
+   when the ID token expires. The
+   expiration date of the Refresh token could be set to the maximum of 30 days.
+1. [CXG9] cellxgene must continue to support the desktop stand alone version
+   and the WSGI hosted version. Authentication is a new feature that can be
+   optionally configured for cellxgene
+   deployments. If cellxgene is not configured for authentication then the login/logout buttons
+   will not appear, and feature such as user-annotations will be controlled as they have been
+   with configuration parameters (e.g. “user_annotations.enable”).
+1. [CXG10] The cellxgene client code must not require a particular solution
+   (i.e. Auth0). Auth0 code must not be added directly to the cellxgene client.
+   Any such specialize code must be
+   introduced using the modularity features of cellxgene. (In the design presented here there is
+   no need to have such specializations in the cellxgene client).
+1. [CXG11] The cellxgene server must not require any authentication specific
+   modules to be installed if authentication is not configured. The cellxgene
+   server may provide Auth0
+   support as a configuration option, but anything specific to the deployment
+   (client_id, client_secret, etc), must be configurable and not part of the code base.
+
+Data Portal
+
+1. [C1] An authenticated user will have the option to logout
+1. [C2] An unauthenticated user will have an option to login
+
+Single sign on
+
+1. [S1] A user authenticated in cellxgene will automatically be authenticated by Data Portal
+1. [S2] A user authenticated in Data Portal will automatically be authenticated by cellxgene
+1. [S3] Logging out of one app, logs the user out of both apps
+
+General
+
+1. [G1] GDPR requirements: For gdpr, the main thing we need to log is user id + time stamp for
+   when they first agreed to our terms of service. The underlying requirement is we need to log
+   evidence of consent, ie the "click" of agreement to our tos.
+1. [G2] The login experience requires that the user has consented to use cookies. It is assumed
+   that this question has already been asked, and we first verify the answer before proceeding
+   with the login process.
+
+## Detailed Design | Architecture | Implementation
```

Comment:
Sorry I forget, did we add any tables to the RDS for cellxgene to keep track of users? Or do we only store their identify for annotations?